### ISSUE TITLE
[error] failed to initialize database, got error dial tcp 127.0.0.1:3306: connect: connection refused

### ISSUE THREAD

**Issue by sanderjo** [ucrawler/nntplexer#2]
```
sander@penguin:~$ ./nntplexer 
2024/07/06 08:04:13 nntplexer v0.0.3 is starting...
2024/07/06 08:04:13 listening on [::]:9998

2024/07/06 08:04:13 /root/v0.0.3/nntplexer.go:226
[error] failed to initialize database, got error dial tcp 127.0.0.1:3306: connect: connection refused
2024/07/06 08:04:13 dial tcp 127.0.0.1:3306: connect: connection refused
sander@penguin:~$ 
```

It would be nice to have an instruction what is needed as database. Just install MySQL, and you're done? Or configuration needed.

And why a database? For caching? No transparent proxy (without database) possible?

EDIT: I installed mysql-server, but same error. Note: I know nothing about MySQL.

**Comment by ucrawler** [comment]
Edit the mysql connection string in nntplexer.ini like this: 

[username[:password]@][protocol[(address)]]/dbname[?param1=value1&...&paramN=valueN]

The database has a backends and users table. You can edit, add and remove on the fly without restarting nntplexer.

Tables are created automatically if they don't exist yet upon starting nntplexer.

Nginx is used on the external side as a ssl offloader.

Regarding your error, you need to create the database before starting nntplexer. 
- mysql
- CREATE database "nntp";
